frontend/javascripts/admin/dataset/dataset_add_zarr_view.tsx

```diff
@@ -0,0 +1,217 @@
+import { Form, Input, Button, Col, Row, Divider } from "antd";
+import { connect } from "react-redux";
+import React, { useState } from "react";
+import type { APIUser } from "types/api_flow_types";
+import type { OxalisState } from "oxalis/store";
+import { exploreRemoteDataset, storeRemoteDataset } from "admin/admin_rest_api";
+import messages from "messages";
+import { jsonStringify } from "libs/utils";
+import { CardContainer } from "admin/dataset/dataset_components";
+import Password from "antd/lib/input/Password";
+import TextArea from "antd/lib/input/TextArea";
+import { AsyncButton } from "components/async_clickables";
+import Toast from "libs/toast";
+import DataLayer from "oxalis/model/data_layer";
+import _ from "lodash";
+import { Hint } from "oxalis/view/action-bar/download_modal_view";
+const FormItem = Form.Item;
+
+type OwnProps = {
+  onAdded: (arg0: string, arg1: string) => Promise<void>;
+};
+type StateProps = {
+  activeUser: APIUser | null | undefined;
+};
+type Props = OwnProps & StateProps;
+
+function DatasetAddZarrView(props: Props) {
+  const { activeUser, onAdded } = props;
+  const [datasourceConfig, setDatasourceConfig] = useState<string>();
```

Comment:
Is there a specific reason for this? If datasource can be null typechecking complains that it shouldn't be used as value for the TextArea.

Reply:
ah, I thought, it should be able to hold `null` since you didn't provide a default value (therefore, it will be null initially). however, if you didn't intend this, you should change it to

```suggestion
  const [datasourceConfig, setDatasourceConfig] = useState<string>("");
```